packages/web/components/Dashboard/LatestContentTabs/Listen.tsx

```diff
@@ -0,0 +1,101 @@
+import { Box, Heading, LoadingState, Text } from '@metafam/ds';
+import React, { useEffect, useState } from 'react';
+import { parse } from 'rss-to-json';
+import useSWR from 'swr';
+
+export const Listen: React.FC = () => {
+  const [items, setItems] = useState<
+    Array<{
+      title: string;
+      src: string;
+      showMore: boolean;
+      description: string;
+    }>
+  >([]);
+
+  const [isLoading, setIsLoading] = useState(true);
+
+  const { data, error } = useSWR(
+    'https://anchor.fm/s/57a641c/podcast/rss',
+    parse,
+  );
+
+  if (error) {
+    console.log(error);
+    setIsLoading(false);
+  }
+
+  useEffect(() => {
+    if (data) {
+      data.items = data.items.map((item) => ({
+        title: item.title,
+        description: item.description.replace(/<\/?[^>]+(>|$)/g, ''),
+        src: item.enclosures[0].url,
+        showMore: false,
+      }));
+
+      setIsLoading(false);
+    }
+  }, [data]);
+
+  const findAndReplace = (src: string) => {
+    const foundIndex = items.findIndex((x) => x.src === src);
+    items[foundIndex].showMore = !items[foundIndex].showMore;
+    return items;
+  };
```

Comment:
```suggestion
  const toggleVisibility = ({ items, member }) => {
    const index = items.findIndex((x) => x.src === member);
    items[index].showMore = !items[index].showMore;
    return items;
  };
```

Reply:
i couldn't go exactly for your solution, because i was getting `Assignment to property of function parameter 'items'.eslintno-param-reassign` at `items[index].showMore = !items[index].showMore;`, so I made a slight adjustment:
```
setPodcasts((items) =>
  toggleVisibility({
    items, <-- no need to make a new array here, because it's made in toggleVisibility
    src: podcast.src,
  }),
)
```
and `toggleVisibility`:
```
const itemsCopy = [...items];
const index = items.findIndex((item) => item.src === src);
itemsCopy[index].showMore = !items[index].showMore;
return itemsCopy;
```